extension/lib/shopify/collection-repository/command/ListGraphQl.js

```diff
@@ -0,0 +1,70 @@
+// const query = require('./Get-GraphQL-query')
+
+class ShopifyCollectionRepositoryCommandListGraphQL {
+  /**
+   * @param {ShopifyStorefrontClient} client
+   */
+  constructor (client) {
+    this._storefrontClient = client
+  }
+
+  /**
+   *
+   * @param {string} id
+   * @returns {Promise<ShopifyCollectionRepositoryCommandGetOutput[]>}
+   */
+  async execute () {
+    console.log('doing execute in listGraphQl')
+    const query = this._storefrontClient.query((root) => {
+      root.add('shop', shop => {
+        shop.add('collections', {
+          args: {
+            first: 250
```

Comment:
What happens if there are more than 250 categories?

Reply:
I am having trouble figuring how to do this. I was able to get the node cursor so I could use it as the after value. I tried setting up a function that would call itself recursively unit the hasNextPage value was false. I haven't been able to get this to work yet. I found an example that uses fetchNextPage works. this worked with the product query used in the shopify example but not the collection query. The error Cannot read property 'shop' of undefined, or  Cannot read property 'nextPageQueryAndPath' of undefined. I am open to any suggestions you have. Otherwise I will continue to work on it tomorrow.